libsolutil/CommonData.h

```diff
@@ -450,6 +450,32 @@ bool contains_if(T const& _t, Predicate const& _p)
 	return std::end(_t) != std::find_if(std::begin(_t), std::end(_t), _p);
 }
 
+/// Function that insert @param _elem into a vector @param _destination only if _elem is not
+/// already present in _destination vector.
+template <class T>
+void tryEmplaceBack(std::vector<T>& _destination, T _elem)
+{
+	if (!contains(_destination, _elem))
+		_destination.emplace_back(std::move(_elem));
+}
+
+/// Function that takes vector @param _destination and appends vector @param _source,
+/// whilst only taking into account unique elements of _source.
+template <class T>
+void appendMissing(std::vector<T>& _destination, std::vector<T> const& _source)
+{
+	for (auto&& elem: _source)
+		tryEmplaceBack(_destination, elem);
+}
+
+/// Function that removes all elements present in vector @param _sub from vector @param _super.
+template <class T>
+void removeVectorSubset(std::vector<T>& _super, std::vector<T> const& _sub)
```

Comment:
Why not make it a `-=` operator for `vector`? We already have `-=` for `set` and `+=` for `vector` here so it would fit the pattern.

Reply:
Yeah, I thought about it, but considering that `+=` doesn't remove duplicates, and that conversely `-=` wouldn't do so either, it seemed a bit odd. I.e., with a `set`, all elements are guaranteed to be unique, whereas in a vector they're not.

Reply:
I don't think this has anything to do with whether `+=` removes duplicates. For that operator just appending items, without considering whether they're duplicates or not, seems like the intuitive default action.

In the same vein leaving the vector in a state where no copy of removed items remains seems like the intuitive default action for `-=`. It's what I'd normally expect and it does not need any extra settings so it's perfect for an operator.

Removing only some copies of specified items seems more like a special case and in that case I'd probably also want to specify which copies to preserve, so a function makes more sense for it.